kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigStateConfigMap.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.kubernetes.operator;
+
+import java.time.Clock;
+import java.util.Comparator;
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.CRUDKubernetesDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
+
+import io.kroxylicious.kubernetes.api.common.Condition;
+import io.kroxylicious.kubernetes.api.common.LocalRef;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.operator.model.ProxyModel;
+import io.kroxylicious.kubernetes.operator.model.ingress.IngressConflictException;
+import io.kroxylicious.kubernetes.operator.model.ingress.ProxyIngressModel;
+import io.kroxylicious.kubernetes.operator.resolver.ResolutionResult;
+
+import static io.kroxylicious.kubernetes.operator.Labels.standardLabels;
+import static io.kroxylicious.kubernetes.operator.ResourcesUtil.namespace;
+
+/**
+ * Generates a Kube {@code ConfigMap} containing some internal state related to proxy configuration.
+ */
+@KubernetesDependent
+public class ProxyConfigStateConfigMap
```

Comment:
To make sure I've got this, we use a distinct configmap containing the conditions. This will provide a reliable ConfigMap for the virtual kafka cluster reconciler to use for it's status regardless of what's going on with the proxy config ConfigMap and we can choose to materialize or delete the proxy config as we please.

Reply:
> To make sure I've got this, we use a distinct configmap containing the conditions. This will provide a reliable ConfigMap for the virtual kafka cluster reconciler to use for it's status regardless of what's going on with the proxy config ConfigMap and we can choose to materialize or delete the proxy config as we please

Exactly